hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestCallerContext.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ipc;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestCallerContext {
+  @Test
+  public void testBuilderAppend() {
+    CallerContext.Builder builder = new CallerContext.Builder("context1");
+    CallerContext context =
+        builder.append("context2").append("key3", "value3", ":").build();
```

Comment:
Having to pass the separator for the key value is weird.

Reply:
Modify the UT.
BTW, we aim to append callercontext and log it, but we don't retrieve item from context. Is it right?
If we want to retrieve item from context, maybe we should use data structure which is serializable and deserializable.